test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -4247,45 +4250,69 @@ def _load_configuration(self, infile, objtype=None):
-                    if ky not in conf:
-                        conf[ky] = {}
-                    newconf = dict(newconf.items() + conf[ky].items())
-                conf = newconf
-
+            with open(infile, 'r') as f:
+                conf = json.load(f)
+            qmgr = os.path.join(self.client_conf['PBS_EXEC'],
+                                'bin', 'qmgr')
             for k, v in conf.items():
-                fn = self.du.create_temp_file()
-                with open(fn, 'w') as fd:
-                    # handle server data saved as output of qmgr commands
-                    # by piping data back into qmgr
-                    if k.startswith('qmgr_'):
-                        qmgr = os.path.join(self.client_conf['PBS_EXEC'],
-                                            'bin', 'qmgr')
-                        fd.write("\n".join(v))
-                        self.du.run_cmd(self.hostname, [qmgr], cstdin=fd,
-                                        sudo=True)
-                    else:
-                        fd.write("\n".join(v))
-                        # append the last line
-                        fd.write("\n")
-                        self.du.run_cmd(self.hostname, ['cp', fn, k],
-                                        sudo=True)
-                    os.remove(fn)
-
+                # Load server configuration
+                if k.startswith('qmgr_'):
+                    fpath = self.du.create_temp_file()
+                    print_svr = '\n'.join(v)
+                    with open(fpath, 'w') as f:
+                        f.write(print_svr)
+                    file_qmgr = open(fpath)
+                    cmd = [qmgr]
+                    d = self.du.run_cmd(
+                        self.hostname, cmd, stdin=file_qmgr, sudo=True)
+                    file_qmgr.close()
+                # Load pbs.conf file
+                elif k == "pbs_conf":
+                    cfg_vals = base64.decodestring(v)
+                    config = ast.literal_eval(cfg_vals)
+                    self.du.set_pbs_config(self.hostname, confs=config)
+                # Load hooks
+                elif k == "hooks":
+                    for hpath in conf['hooks']:
+                        h_name = hpath.split(os.sep)[-1]
+                        hook_name = str(h_name[:-3])
+                        # Skip loading PBS hook
+                        if hook_name.startswith("PBS_"):
+                            continue
+                        cf_hpath = ""
+                        if ".PY" in hpath:
+                            cf_hpath = hpath.replace('.PY', '.CF', 1)
+                            hk_hpath = hpath.replace('.PY', '.HK', 1)
+                            rep_str = "hook_name=" + hook_name
+                            cmd = qmgr + " -c 'create hook %s'" % hook_name
+                            self.du.run_cmd(
+                                self.hostname, cmd, sudo=True,
+                                as_script=True)
```

Comment:
This is just a question. why do you need as_script=True? Did you try making it work without that?